### PR TITLE
Development

### PR DESCRIPTION
**Pull request by andleebsyed**
Hey Guys  , Welcome to the PR of my ecommerce app, 
please feel free to pass on  the feedbacks or suggestions if any
Thanks.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+@media (prefers-reduced-motion: no-preference) {
+  .App-logo {
+    animation: App-logo-spin infinite 20s linear;
+  }
+}
+
+.App-header {
+  background-color: #282c34;
```
Comment (shanudjn) [inline src/App.css:29]: Extract colors to a variable in a seprate css file

File: src/components/Account/Account.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    if (user) {
+        return (
+            <div className="outer-div">
+                <div className="heading">
+                    <h1 className="welcome-text">{user.username}</h1>
+                    <button className="button button-primary button-custom" onClick={handleLogout}>Logout</button>
+                </div>
+                <div className="cards-container">
+
+                    <ProfileDetails />
```
Comment (shanudjn) [inline src/components/Account/Account.js]: Liked how you seperated the code into smaller components

File: src/components/DatabaseCalls/DatabaseCalls.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// import { useContext, createContext, useEffect, useState } from "react";
```
Comment (shanudjn) [inline src/components/DatabaseCalls/DatabaseCalls.js]: Database.js is commented. It can be removed

File: src/components/Signup/Signup.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                <input type="submit" className="button button-outline login-button" value="Sign In" />
+                <p>Already have an account? <Link className="signup-link" to='/account'>Log in</Link></p>
+            </div>
+        </form>
+    )
+}
+    // }
+
+
+    // export function LoginProvider({ children }) {
```
Comment (shanudjn) [inline src/components/Signup/Signup.js]: commented code can be removed

**Review by shanudjn — COMMENTED** [review]
Hey Andleeb, very well written code. I was very readable. Have commented some suggestions. Please let me know if any concerns

File: src/components/Signup/Signup.js (hunk 4 above)
Comment (andleebsyed) [reply src/components/Signup/Signup.js]: done

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+      <Routes>
+        <Route path="/" element={<div className="homepage"><Homepage /></div>} />
+        <Route path="/products" element={<div className="products"><Products /></div>} />}
+
+        <PrivateRoute path="/wishlist" element={<div className="wishlist"><Wishlist /></div>} />
+        {/* <PrivateRoute path="/products" element={<div className="products"><Products /></div>} />} */}
+
+        <PrivateRoute path="/cart" element={<div className="cart"><Cart /></div>} />
+        <PrivateRoute path="/search" element={<div className="searched-products"><Search /></div>} />
```
Comment (Himanshu916) [inline src/App.js]: Hey i think you should not make this search component private .

File: src/components/ecom-context/ecom-context.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    switch (type) {
+        case 'INITIAL_PRODUCTS':
+            return { ...state, data: [...payload], orgData: [...payload] }
+        case 'INITIAL_CART':
+            return { ...state, cart: [...payload] }
+        case 'INITIAL_WISHLIST':
+            return { ...state, wishlist: [...payload] }
+        case 'ADD_TO_WISHLIST':
+            return { ...state, cart: cart.filter(item => item._id !== payload._id), wishlist: [...wishlist, payload] }
+        case 'ADD_TO_CART':
```
Comment (Himanshu916) [inline src/components/ecom-context/ecom-context.js]: when the data is adding into cart, is it not adding to backend also ? and then fetched it from there?
similarly for wishlist  . If not then i think it should be added to backend first and then set it to state.

**Review by Himanshu916 — CHANGES_REQUESTED** [review]
Hey Syed ! I have suggested you some changes in your app.js file and Ecomm context . Can you please look into them

File: src/App.js (hunk 5 above)
Comment (andleebsyed) [reply src/App.js]: thanks . was working on it , made public now

File: src/components/ecom-context/ecom-context.js (hunk 6 above)
Comment (andleebsyed) [reply src/components/ecom-context/ecom-context.js]: it is being added to the backend , for reference you can check components/utils/Operations.js. there first data gets added to db and only when that returns success the particular dispatch is fired.

File: src/App.css (hunk 1 above)
Comment (andleebsyed) [reply src/App.css:29]: done brother, but not to a separate file but kept variable global in the same css file and used that wherever needed

File: src/components/DatabaseCalls/DatabaseCalls.js (hunk 3 above)
Comment (andleebsyed) [reply src/components/DatabaseCalls/DatabaseCalls.js]: removed

**Pull request merged**